### PR TITLE
Add pyproject.toml and packaging config

### PR DESCRIPTION
**Pull request by luizirber**
Add pyproject.toml with setuptools build backend, and use `uv` to verify package works.

- https://github.com/astral-sh/setup-uv
- https://docs.astral.sh/uv/

Likely review this after #9 is merged, and how the `requirements.txt` file is generated

### CODE REVIEW

**Comment by jeff-cohere** [comment]
@ialarmedalien , Luiz Irber is one of our BER Data Integration Shenanigans colleagues, and another contractor type hired by Kjiersten to Do Stuff. He believes in reducing friction, and has a bit more familiarity with the miasma of Python-related package management tools. Can you comment on how uv, etc, fits in with other KBase tools of interest?

**Comment by jeff-cohere** [comment]
The tests are failing for this PR because the `DTS_KBASE_DEV_TOKEN` secret isn't available to PRs from a fork.

**Comment by ialarmedalien** [comment]
General point: let's just standardise to using py3.12 instead of having 3.11 in some places and 3.12 in others.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+description = "Python client for the Data Transfer Service"
+authors = [
+  { name = "Jeffrey N. Johnson", email = "jeff@cohere-llc.com" },
+]
+readme = "README.md"
+requires-python = ">=3.12"
+dependencies = [
+  "frictionless>=5.17.0,<6",
+  "requests>=2.32.3,<3",
+]
```
Comment (ialarmedalien) [inline pyproject.toml:17]: have you tried running dtspy from a clean py3.12 install? The requirements file is quite long - are these the only two deps? I hope that the testing and doc generation aren't responsible for all the rest!

**Comment by ialarmedalien** [comment]
We need either an additional PR or amendments to this PR that do the following:

- separate out the deps into runtime, testing, and documentation generation -- in particular, there seem to be a ton of documentation things that people are not going to need if they run the code locally
- additional information in the README about how to install and run the project locally -- currently there's no documentation at all and only people who are already python-savvy or who want to pick through the GitHub actions will work out how to do it.

@luizirber we appreciate your contribution and don't want to dump a load of extra work on you so can you and @jeff-cohere  divide and conquer those tasks between yourselves?

File: pyproject.toml (hunk 1 above)
Comment (luizirber) [reply pyproject.toml:17]: These two are the top-level dependencies, most of the ones listed in the requirements file come from them indirectly.

To see the equivalent `requirements.txt` based on the uv lock file, you can run
`uv pip compile pyproject.toml`
(and maybe add `--no-annotate` to make it easier to diff with current `requirements.txt`)

But yes, there are many that are testing/doc deps...

**Comment by luizirber** [comment]
> We need either an additional PR or amendments to this PR that do the following:
> 
>     * separate out the deps into runtime, testing, and documentation generation -- in particular, there seem to be a ton of documentation things that people are not going to need if they run the code locally

This is somewhat done already, dev deps are listed here:
https://github.com/kbase/dtspy/pull/10/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R19-R26

I can add another level to be doc vs testing deps

>     * additional information in the README about how to install and run the project locally -- currently there's no documentation at all and only people who are already python-savvy or who want to pick through the GitHub actions will work out how to do it.

Agreed, will add to this PR!

> @luizirber we appreciate your contribution and don't want to dump a load of extra work on you so can you and @jeff-cohere divide and conquer those tasks between yourselves?

:+1:

File: pyproject.toml (hunk 1 above)
Comment (luizirber) [reply pyproject.toml:17]: And I'm already using as a dep in a project with
```
[project]
dependencies = [
  "dtspy @ https://github.com/luizirber/dtspy/archive/724f2884e14fee4cc9f9019e3bdf63a46aecf487.tar.gz"
]
```
which points to this PR, and it is all working

**Comment by jeff-cohere** [comment]
Thanks, both. I find the Python and Javascript communities to be similarly ultra-aggressive in terms of dependencies, and I haven't really formed my own philosophy about how to manage the resuting pain that these communities feel is a cost of doing business.

@luizirber , thanks for taking point on this particular topic for the moment. Happy to kick ideas around.

**Comment by jeff-cohere** [comment]
> additional information in the README about how to install and run the project locally -- currently there's no documentation at all and only people who are already python-savvy or who want to pick through the GitHub actions will work out how to do it.

This is on my list for the very next PR!

**Comment by ialarmedalien** [comment]
Separate deps segments for mkdocs and testing would be great, thanks. Something super simple would be fine for the readme -- @jeff-cohere can always flesh it out in his upcoming PR.

**Review by ialarmedalien — APPROVED** [review]

**Pull request merged**